sdk/openai/openai/test/public/openai.spec.ts

```diff
@@ -212,12 +211,36 @@ describe("OpenAI", function () {
                   chatCompletionDeployments,
                   chatCompletionModels
                 ),
-                async (deploymentName) =>
-                  client
-                    .getChatCompletions(deploymentName, weatherMessages, {
-                      functions: [getCurrentWeather],
-                    })
-                    .then((c) => assertChatCompletions(c, { functions: true }))
+                async (deploymentName) => {
+                  const weatherMessages: ChatMessage[] = [
+                    { role: "user", content: "What's the weather like in Boston?" },
+                  ];
+                  const result = await client.getChatCompletions(deploymentName, weatherMessages, {
+                    functions: [getCurrentWeather],
+                  });
+                  assertChatCompletions(result, { functions: true });
+                  const responseMessage = result.choices[0].message;
+                  if (responseMessage?.functionCall) {
+                    const functionArgs = JSON.parse(responseMessage.functionCall.arguments);
```

Comment:
Should these be deserialized by the client?

Reply:
Yes, because we want to extend the messages with the function call response